karapace/backup/api.py

```diff
@@ -282,9 +282,8 @@ def _consume_records(
     consumer: KafkaConsumer,
     topic_partition: TopicPartition,
     poll_timeout: PollTimeout,
-) -> Iterator[ConsumerRecord]:
-    start_offset: int = consumer.beginning_offsets([topic_partition])[topic_partition]
-    end_offset: int = consumer.end_offsets([topic_partition])[topic_partition]
+) -> Iterator[Message]:
+    start_offset, end_offset = consumer.get_watermark_offsets(topic_partition)
```

Comment:
I had another doubt this was correct due to mix of terminology, but have now verified it's the same underlying Protocol message involved (`ListOffsets`):

- https://github.com/confluentinc/librdkafka/blob/df6efd25e6e44c06a937cbdf8fc81715668e0ce1/src/rdkafka.c#L3597
- https://github.com/dpkp/kafka-python/blob/0864817de97549ad71e7bc2432c53108c5806cf1/kafka/consumer/fetcher.py#L573

So looks correct 👍